webui/react/src/hooks/usePolling.ts

```diff
@@ -28,6 +28,12 @@ const DEFAULT_OPTIONS: PollingOptions = {
   runImmediately: true,
 };
 
+/**
+ * Polling hook that polls a given polling function.
+ * @param pollingFn
+ *    The function to poll. Pass an async/await function to ensure
```

Comment:
suggestion: if we only ever usePolling for async functions, or want to enforce, that wdy think about adding a runtime check that the passed in function actually does return a promise?

typescript types can probably also check for the declared return type of the `pollingFn`
`type PollingFn = (() => Promise<void>) | (() => void);`

Reply:
This is a good suggestion. I thought about this but decided to hold off on this because there are some polling calls that are non-async based still. But it's possible that they go away when another big PR (task logs) lands. Then at that point, we can enforce the polling function to be async functions.